src/components/App/component.tsx

```diff
@@ -44,39 +44,42 @@ const App = () => {
   const [projectId] = useLocalStorage("patchProjectId", "new");
   const { loadProject } = useProjectActions(projectId);
   const onVmInit = () => {
-      loadProject();
+    loadProject();
   }
   useInitializedVm(onVmInit);
 
   useThreadAutoSave(patchVM, saveTargetThreads, editorTab);
   useMonitorProjectChange(setProjectChanged, [targetIds])
-  
+
 
   return (
     <ThemeProvider theme={mode === "dark" ? darkTheme : lightTheme}>
-      <SplashScreen renderCondition={true}>
-        <ToastContainer
-          theme="dark"
-          position="top-center"
-         />
-        <Grid container item direction="row" width={'100%'} sx={{
-          position: "absolute",
-          width: "100%",
-          top: 0,
-          left: 0,
-          right: 0,
-          bottom: 0,
-          padding: 0,
-          margin: 0,
-          paddingBottom: "0px",
-          zIndex: -1,
-          overflowY: "auto",
-          backgroundColor: 'background.default',
-          color: "text.primary",
-        }}>
-          <ModalSelector />
-          <TopBar mode={mode} setMode={setMode} />
-          <Grid item container direction="row" className="leftContainer">
+      {!patchVM ?
+        <SplashScreen renderCondition={false} />
```

Comment:
The splash screen component conditionally renders it's children. So, we probs can refactor ternary into the SplashScreen renderCondition parameter like such
```
<SplashScreen renderCondition={!!patchVM}>
// rest of code...
<SplashScreen/>
```
Additionally, we need to include other conditions in the renderCondition parameter. Primarily, there is an event fired from the VM (the vm extends the event emitter class) when a project is fully loaded into the vm. This is when we should render the ide in full.

Reply:
Okay, I looked through the VM codebase. There is a "WORKER READY" event that you should listen for. But you may need to make a PR for the VM, to emit a "PROJECT_LOADED" event then listen for that as well in the ide